ckan/cli/search_index.py

```diff
@@ -77,6 +77,43 @@ def clear(dataset_name: str):
         clear_all()
 
 
+def list_orphans(return_list: bool=False) -> list|None:
+    import ckan.logic as logic
+    search = logic.get_action('package_search')({},{
+                'q': '*:*',
+                'fl': 'id'})
+    indexed_package_ids = {r['id'] for r in search['results']}
+
+    import ckan.model as model
+    from sqlalchemy.sql import select
+    package_ids = {r[0] for r in select([model.package_table.c['id']]).execute()}
```

Comment:
* `model.Session.query(model.Package.id)`, just to be consistent with the other SQL queries
* Do you really need to convert the DB cursor into the set? I don't think that you can get duplicated IDs here

Reply:
package_ids is a set here so that line 93 below will be O(1)

Reply:
indexed_package_ids could be a list since we're only iterating over it though :-)

Reply:
`package_ids` remaining as a set, but indexed_package_ids is now just staying as a list and iterating through that.

Reply:
`select([model.package_table.c['id']]).execute()` has been changed to `model.Session.query(model.Package.id)`